lib/api/bulk.js

```diff
@@ -374,12 +377,36 @@ Batch.prototype._createRequestStream = function() {
   var batch = this;
   var bulk = batch._bulk;
   var logger = bulk._logger;
+  var contentType;
+
+  switch(batch.job.options.contentType) {
```

Comment:
You could also compare to enum values here in the switch. And if you're feeling really feisty, you could stack the CSV case right above the default without a break, so that it falls through into the default.